packages/x-tree-view/src/internals/plugins/useTreeViewExpansion/useTreeViewExpansion.types.ts

```diff
@@ -2,10 +2,20 @@ import * as React from 'react';
 import { DefaultizedProps, TreeViewPluginSignature } from '../../models';
 import { UseTreeViewItemsSignature } from '../useTreeViewItems';
 
-export interface UseTreeViewExpansionInstance {
+export interface UseTreeViewExpansionPublicAPI {
+  /**
+   * Change the expansion status of a given item.
+   * @param {React.SyntheticEvent} event The UI event that triggered the change.
+   * @param {string} itemId The id of the item to modify.
+   * @param {boolean} isExpanded The new expansion status of the given item.
+   */
+  setItemExpansion: (event: React.SyntheticEvent, itemId: string, isExpanded: boolean) => void;
```

Comment:
The main DX question here is about the `event`

Do we require the user to pass an event? This is needed right now to call the `onItemExpansionToggle` prop.
If we don't, do we only call `onItemExpansionToggle` when the event is defined or do we make the event nullable on `onItemExpansionToggle`?

Reply:
We could imagine scenarios where there is no event (expand an item when receiving some data from the server for instance).
But I'd be in favor of keeping the event mandatory and making it nullable if we have feedback of valid use-cases that ask for it.